src/gui/plugins/scene3d/Scene3D.hh

```diff
@@ -439,6 +439,16 @@ inline namespace IGNITION_GAZEBO_VERSION_NAMESPACE {
     /// when the deselection is initiated from this plugin.
     private: void DeselectAllEntities(bool _sendEvent);
 
+    /// \brief Helper function that traverses the node tree looking for any
+    /// child with the "geometry-type" user data. If found, the function also
+    /// checks if the value is a simple shape (box, cylinder or sphere).
+    /// \param[in] _node Root node.
+    /// \param[out] _visual The visual pointer to the simple shape if found.
+    /// \return True if any simple shape was found in the tree or false
+    /// otherwise.
+    private: bool ContainsSimpleShape(const rendering::NodePtr &_node,
```

Comment:
Could you return the VisualPtr as `nullptr` if there was no shape, rather than as an arg and as a bool?

Reply:
Thanks! Changed in d449365.